tests/cli/CopyDirRecursiveExcludeCept.php

```diff
@@ -0,0 +1,49 @@
+<?php
+$I = new CliGuy($scenario);
+
+$I->wantTo('copy dir recursively with CopyDir task, but exclude a file');
+$I->amInPath(codecept_data_dir().'sandbox');
+$I->seeDirFound('some/deeply/nested');
+$I->seeDirFound('some/deeply/nested2');
+$I->seeDirFound('some/deeply/nested3');
+$I->seeDirFound('some/deeply/nested3/nested31');
+$I->seeDirFound('some/deeply/nested4');
+$I->seeDirFound('some/deeply/nested4/nested41');
+$I->seeFileFound('structu.re', 'some/deeply/nested');
+$I->seeFileFound('structu1.re', 'some/deeply/nested');
+$I->seeFileFound('structu2.re', 'some/deeply/nested');
+$I->seeFileFound('structu3.re', 'some/deeply/nested');
+$I->seeFileFound('structu21.re', 'some/deeply/nested2');
+$I->seeFileFound('structu31.re', 'some/deeply/nested3');
+$I->seeFileFound('structu32.re', 'some/deeply/nested3');
+$I->seeFileFound('structu311.re', 'some/deeply/nested3/nested31');
+$I->seeFileFound('structu411.re', 'some/deeply/nested4/nested41');
+$I->seeFileFound('structu412.re', 'some/deeply/nested4/nested41');
+$I->taskCopyDir(['some/deeply' => 'some_destination/deeply'])
+    ->exclude([
+        // Basename exclusion.
+        'structu1.re',
+        // File in subdir exclusion.
+        'some/deeply/nested/structu3.re',
+        // Dir exclusion.
+        'nested2',
+        // Subdir exclusion.
+        'some/deeply/nested3/nested31',
+        // Subpath within source exclusion.
+        'nested3/structu31.re',
+        // File in deeper subpath within source exclusion.
+        'nested4/nested41/structu411.re',
```

Comment:
Yep, this is the case I was talking about. Thanks for adding this to the unit tests to clarify.